src/linker/Linker/Driver.cs

```diff
@@ -434,6 +434,12 @@ protected int SetupContext (ILogger customLogger = null)
 
 						continue;
 
+					case "--generate-warning-suppressions":
+						if (!GetStringParam (token, l => context.GenerateWarningSuppressions (l)))
```

Comment:
As is the generated code is "Wrong" - we generate suppressions for all assemblies into one file. But global warning suppressions don't work cross-assembly. So someone would have to go, manually decide which suppression belongs to which assembly (tricky - we only write namespace/type) and split them into their respective assemblies.

I think this should be a simple boolean option and we will write `AssemblyName.WarningSuppressions.cs` for each assembly which has at least one warning to suppress into the output directory.